homeassistant/components/reolink/entity.py

```diff
@@ -13,9 +12,7 @@
 class ReolinkCoordinatorEntity(CoordinatorEntity):
     """Parent class for Reolink Entities."""
 
-    def __init__(
-        self, reolink_data: ReolinkData, config_entry: ConfigEntry, channel: int | None
-    ) -> None:
+    def __init__(self, reolink_data: ReolinkData, channel: int | None) -> None:
```

Comment:
```suggestion
    def __init__(self, reolink_data: ReolinkData, channel: int) -> None:
```

Reply:
No, the channel can actually be None for entities that control the NVR itself as apposed to a camera connected to the NVR.

For instance the firmware update entity of the NVR and switches that control global NVR settings.
(Those entities have not been merged yet, but they will be next).

Reply:
> No, the channel can actually be None for entities that control the NVR itself as apposed to a camera connected to the NVR.

In that case, the typing of this PR is wrong in many places and need to take `None` into account. Additionally, the upstream library doesn't accept `None` for its channel, so the upstream library is either broken or incorrectly typed.

Reply:
For example:

![image](https://user-images.githubusercontent.com/195327/212783349-58a334e0-3562-459d-857b-802376b3f0be.png)

Reply:
@frenck alright, technically you are right.
The upstream lib is fine, motion_detected and many other methods can not be used for the NVR itself (since it is not a camera it cannot detect motion)
So for all **camera** entities the channel parameter is indeed required and cannot be None.
While for a NVR entitie there is no channel parameter so it schould not be None but schould be absend.

I made a commit "[Split out NVR and camera entities](https://github.com/home-assistant/core/pull/85654/commits/1a673c405ed766cec43e1cc9b8c1d6a1a7015dac)" which makes the typing correct (required int). And split of a separete class for the NVR entities.

It does add a lot of code duplication for the single channel can not be None typing, but I guess it is fine.